### PR TITLE
Move convert.py to examples/convert-legacy-llama.py

### PR DESCRIPTION
**Pull request by Galunid**
There's little use for this one, and a lot of confusion.

### CODE REVIEW

**Comment by ggerganov** [comment]
We should update readme instructions, `ggml-ci` and anything else that references the script. "GGUF My Repo" would also need an update: https://huggingface.co/spaces/ggml-org/gguf-my-repo/blob/main/app.py#L29

**Comment by mofosyne** [comment]
No torch?

My understanding is that it's for safetensors

File: docs/HOWTO-add-model.md
Comment (cebtenzzre) [inline docs/HOWTO-add-model.md:20]: Given that .pth *stands for* pytorch, I think this name could be misleading. Why not call it convert-llama.py?

Comment (Galunid) [reply docs/HOWTO-add-model.md:20]: `convert-legacy-llama.py` maybe? I'd like to avoid giving impression that it can be used for converting `llama 3`

**Comment by teleprint-me** [comment]
I always just named it `convert-torch.py` because that's what it did. I mean, if you really want to be clear about it, `convert-torch-to-gguf.py`.

File: examples/convert-no-torch.py
Comment (compilade) [inline examples/convert-no-torch.py:1]: `convert-hf-to-gguf.py` imports should be changed to reflect the rename

https://github.com/ggerganov/llama.cpp/blob/88e06d04e5e93be80565ed957fd2f41c047749ce/convert-hf-to-gguf.py#L27

Comment (teleprint-me) [reply examples/convert-no-torch.py:1]: It should be moved to `vocab.py`.

**Comment by mofosyne** [comment]
Hmmmm... Do we have a conversion reference table so people can identify what file type to use with each script?

If not for code repo docs maybe that's a wiki entry

**Comment by mofosyne** [comment]
Not against this idea as convert.py is indeed a bit confusing, but this PR is still appearing broken with the CI.

I do wonder if we could somehow consolidate all these into one convert.py... and have convert.py call specific convert-*.py based on detection of the model source etc...

**Comment by mofosyne** [comment]
If you decide to go though the consolidation approach, see if you can also consolidate some aspect like how we generate [default naming convention gguf](https://github.com/ggerganov/ggml/blob/master/docs/gguf.md#gguf-naming-convention).

This will make it easier to update it as the standards evolve.

File: gguf-py/gguf/vocab.py
Comment (teleprint-me) [inline gguf-py/gguf/vocab.py:19]: I'm wondering how to handle these. Having floating references, even as constants, seems questionable. This is fine for now until we figure out how to deal with these.

Comment (Galunid) [reply gguf-py/gguf/vocab.py:19]: Can we just replace them with explicitly using strings? I think they are used only in one/two places. I just copied them over from `convert.py`, but I missed that `gguf` exports them.

**Comment by Galunid** [comment]
> Vocab factory is missing too. Not sure how to handle that yet.

I don't think there's any use for vocab factory in `convert-hf-to-gguf.py`, nor any other scripts. I think it's fine to leave it in `convert-llama-legacy.py` since that's the only place we can expect to use it.

File: requirements/requirements-convert-legacy-llama.txt
Comment (compilade) [inline requirements/requirements-convert-legacy-llama.txt:1]: There's a `requirements.txt` file in the `llava` example which still refers to `../../requirements/requirements-convert.txt`

File: examples/convert-legacy-llama.py
Comment (compilade) [inline examples/convert-legacy-llama.py:1]: In `.devops/tools.sh`, there's one last reference to `convert.py` (which could probably be replaced by `convert-hf-to-gguf.py`). I think it's the script used as the entrypoint in Docker containers.

**Review by compilade — APPROVED** [review]
Looks good to me. But I think <https://huggingface.co/spaces/ggml-org/gguf-my-repo/discussions/68> should be merged first to avoid breaking that HF space.

@ggerganov, @phymbert, @Vaibhavs10

**Comment by teleprint-me** [comment]
Awesome! I've been waiting for this to be merged so I can integrate this into my PR. This will be incredibly useful so I don't need to duplicate or reinvent APIs.

**Review by mofosyne — APPROVED** [review]
Hoping this wont break my PR https://github.com/ggerganov/llama.cpp/pull/7499#pullrequestreview-2075739095 

Looks like a good idea and good to refactor at this stage. Let's merge in once the hugging face PR gets merged in.

---

But let's merge if hf have not merged theirs in still by the time the main branch is stable.

**Comment by Vaibhavs10** [comment]
Heya! @mofosyne, @Galunid and @ggerganov - I just merged a PR (https://huggingface.co/spaces/ggml-org/gguf-my-repo/discussions/73) to default to the convert-hf script.

Good from my side to merge this!

**Review by Vaibhavs10 — APPROVED** [review]
Also, thanks a ton for taking this up and opening the PR! 🤗

**Review by mofosyne — APPROVED** [review]
Well, it's now or never. Let's see how many convert.py PRs we have to fix, but looking forward to a cleaner codebase.

**Pull request merged**
